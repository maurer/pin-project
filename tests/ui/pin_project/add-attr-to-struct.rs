use std::marker::PhantomPinned;

use auxiliary_macro::add_pin_attr;
use pin_project::pin_project;

#[pin_project]
struct Foo {
    #[pin]
    #[pin] //~ ERROR duplicate #[pin] attribute
    f: PhantomPinned,
}

#[add_pin_attr(struct)] //~ ERROR #[pin] attribute may only be used on fields of structs or variants
#[pin_project]
struct Bar {
    #[pin]
    f: PhantomPinned,
}

fn main() {}
