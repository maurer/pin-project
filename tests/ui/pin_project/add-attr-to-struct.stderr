error: duplicate #[pin] attribute
 --> tests/ui/pin_project/add-attr-to-struct.rs:9:5
  |
9 |     #[pin] //~ ERROR duplicate #[pin] attribute
  |     ^^^^^^

error: #[pin] attribute may only be used on fields of structs or variants
  --> tests/ui/pin_project/add-attr-to-struct.rs:13:1
   |
13 | #[add_pin_attr(struct)] //~ ERROR #[pin] attribute may only be used on fields of structs or variants
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `add_pin_attr` (in Nightly builds, run with -Z macro-backtrace for more info)
